app/src/org/commcare/navdrawer/BaseDrawerActivity.kt

```diff
@@ -61,8 +62,20 @@ abstract class BaseDrawerActivity<T> : CommCareActivity<T>() {
         ) { success: Boolean ->
             if (success) {
                 goToConnectJobsList(this)
-                setResult(RESULT_OK)
-                finish()
+                drawerController?.closeDrawer()
+            }
+        }
+    }
+
+    private fun navigateToMessaging() {
+        val personalIdManager: PersonalIdManager = PersonalIdManager.getInstance()
+        personalIdManager.init(this)
+        personalIdManager.unlockConnect(
+            this
+        ) { success: Boolean ->
+            if (success) {
+                goToMessaging(this)
+                drawerController?.closeDrawer()
```

Comment:
think the drawer logic should live out of this method so that we can move this method to `ConnectNavHelper` as well